core/sqf/src/seatrans/tm/hbasetmlib2/src/main/java/org/trafodion/dtm/HBaseTxClient.java

```diff
@@ -548,24 +538,31 @@ public short doCommit(long transactionId) throws Exception {
           if (useTlog) {
              tLog.putSingleRecord(transactionId, commitIdVal, "COMMITTED", ts.getParticipatingRegions(), true);
           }
-       } catch(Exception e) {
-          LOG.error("Returning from HBaseTxClient:doCommit, txid: " + transactionId + " tLog.putRecord: EXCEPTION " + e);
+       } catch(IOException e) {
+          LOG.error("Returning from HBaseTxClient:doCommit, txid: " + transactionId + " tLog.putRecord: EXCEPTION ", e);
           return TransReturnCode.RET_EXCEPTION.getShort();
        }
 
        if ((stallWhere == 2) || (stallWhere == 3)) {
           LOG.info("Stalling in phase 2 for doCommit");
-          Thread.sleep(300000); // Initially set to run every 5 min
+          boolean loopBack = false;
+          do {
+             try {
+                Thread.sleep(300000); // Initially set to run every 5 min
```

Comment:
Should there be a loopBack = false after sleep in the case where an exception occurred but then the operation succeeded?